src/commands/getrange.rs

```diff
@@ -0,0 +1,204 @@
+use bytes::Bytes;
+use std::sync::{Arc, Mutex};
+use std::usize;
+
+use crate::commands::executable::Executable;
+use crate::commands::{CommandParser, CommandParserError};
+use crate::frame::Frame;
+use crate::store::Store;
+use crate::Error;
+
+/// Returns the substring of the string value stored at key, determined by the offsets start and
+/// end (both are inclusive). Negative offsets can be used in order to provide an offset starting
+/// from the end of the string. So -1 means the last character, -2 the penultimate and so forth.
+/// The function handles out of range requests by limiting the resulting range to the actual length
+/// of the string.
+///
+/// Ref: <https://redis.io/docs/latest/commands/getrange/>
+#[derive(Debug, PartialEq)]
+pub struct Getrange {
+    pub key: String,
+    pub start: i64,
+    pub end: i64,
+}
+
+impl Executable for Getrange {
+    fn exec(self, store: Arc<Mutex<Store>>) -> Result<Frame, Error> {
+        let store = store.lock().unwrap();
+        let value = store.get(&self.key);
+        let bytes = match value {
+            Some(val) => val,
+            None => return Ok(Frame::Bulk(Bytes::new())),
+        };
+
+        let value = String::from_utf8(bytes.to_vec()).unwrap();
+        // TODO: Should we worry about this conversion?
+        let len = value.len() as i64;
```

Comment:
If somebody stores a string that is 92233720368547 characters long we have a different problem (if my calculation is not wrong it'd be ~8000 Petabytes).

Reply:
😁 we should add a frame limit size, so we would never have these kind of issues. That logic I think should live in the codec.